src/nanoemoji/write_font.py

```diff
@@ -273,26 +274,38 @@ def _update_paint_glyph(paint):
         reuse_result = glyph_cache.try_reuse(path_in_font_space)
         if reuse_result is not None:
             # TODO: when is it more compact to use a new transforming glyph?
+            child_transform = Affine2D.identity()
             child_paint = paint.paint
-            if is_transform(child_paint) and is_gradient(child_paint.paint):
-                # We have a transformed gradient so we need to reverse the effect of the
+            if is_transform(child_paint):
+                child_transform = child_paint.gettransform()
+                child_paint = child_paint.paint
+
+            assert fixed_safe(*reuse_result.transform)
```

Comment:
Provide a message, particularly anything that would inform user what glyph has the problem. I'm not sure if Python follows quite the same conventions but in other languages I'd say this should raise not assert

Reply:
I deliberately made it an assertion instead of an exception because the shape reuse code already ensures that only fixed-safe transforms are used and skips those that would overflow, so I deem this an issue of internal code logic which should never be raised. I can add a comment to clarify that